src/worldcereal/rdm_api/rdm_interaction.py

```diff
@@ -95,54 +102,89 @@ def _get_headers(self) -> Dict[str, str]:
             headers.update(self.headers)
         return headers
 
-    def _collections_from_rdm(
+    def get_collections(
         self,
-        geometry: BaseGeometry,
+        geometry: Optional[BaseGeometry] = None,
         temporal_extent: Optional[List[str]] = None,
-    ) -> List[str]:
-        """Queries the RDM API and finds all intersection collection IDs for a given geometry and temporal extent.
+        include_public: Optional[bool] = True,
+        include_private: Optional[bool] = False,
+        ewoc_codes: Optional[List[int]] = None,
+    ) -> List[RdmCollection]:
+        """Queries the RDM API and finds all intersecting collections
+        for a given geometry and temporal extent.
 
         Parameters
         ----------
-        geometry : BaseGeometry
+        geometry : Optional[BaseGeometry], optional
             A user-defined geometry for which all intersecting collection IDs need to be found.
+            CRS should be EPSG:4326.
+            If None, all available data will be queried., by default None
         temporal_extent : Optional[List[str]], optional
             A list of two strings representing the temporal extent, by default None. If None, all available data will be queried.
+        include_public: Optional[bool] = True
+            Whether or not to include public collections.
+        include_private: Optional[bool] = False
+            Whether or not to include private collections.
+            If True, the user must be authenticated.
+        ewoc_codes: Optional[List[int]] = None
+            A list of EWOC codes to filter the collections by.
+
         Returns
         -------
-        List[str]
-            A List containing the URLs of all intersection collection IDs.
+        List[RdmCollection]
+            A List containing the collections matching the criteria.
         """
 
-        bbox = geometry.bounds
+        # If user requests private collections, they must be authenticated
+        if include_private:
+            if not self.headers or "Authorization" not in self.headers:
+                logger.info("To access private collections, you need to authenticate.")
+                self.authenticate()
+
+        # Handle geometry
+        bbox = geometry.bounds if geometry else [-180, -90, 180, 90]
```

Comment:
probably safer: `bbox = geometry.bounds if geometry is not None else [-180, -90, 180, 90]`.
Do we need a sanity check on CRS, e.g. by checking the bounds ranges for impossible values?